src/utils/processFile.ts

```diff
@@ -0,0 +1,73 @@
+import { AbortError } from '../errors';
+
+/**
+ * @hidden
+ */
+export type onChunkCallback = (chunk: string | ArrayBuffer, offset: number) => void;
+
+/**
+ * @hidden
+ */
+export type processFileOptions = {
+    file: Blob;
+    chunkSize: number;
+    signal?: AbortSignal;
+    onChunkCallback: onChunkCallback;
+    onFinishCallback: () => void;
+    onErrorCallback: (err: any) => void;
+};
+
+/**
+ * @hidden
+ */
+export function processFile({
+    file,
+    chunkSize,
+    signal,
+    onChunkCallback,
+    onFinishCallback,
+    onErrorCallback,
+}: processFileOptions) {
+    const reader = new FileReader();
+
+    const dataSize = file.size;
+
+    let offset = 0;
+    let endOffset = Math.min(offset + chunkSize, dataSize);
+
+    if (signal) {
+        const onAbort = () => {
+            reader.abort();
+            onErrorCallback(new AbortError());
+        };
+        if (signal.aborted) return onAbort();
+        else signal.addEventListener('abort', onAbort);
+    }
+
+    reader.onload = () => {
+        if (!reader.result) throw new SyntaxError('reader.result is null');
```

Comment:
The `SyntaxError` object represents an error when trying to interpret syntactically invalid code [(mdn)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError). Please use regular `Error`